### ISSUE TITLE
[Bug]: Training with speechbrain pretrained models from huggingface repo

### ISSUE THREAD

**Issue by miketauro** [speechbrain/speechbrain#2022]
### Describe the bug

Hi all,

I've used huggingface hosted models like "microsoft/wavlm-large" and whisper to perform some speech classification tasks. This has worked well. Now I would like to use one of my wavlm (or whisper, wav2vec2, etc) models trained for the aforementioned speech classification task to fine-tune on another speech classification task. 
I thought I could do this by uploading the pretrained model "wavlm.ckpt" to a hugginface repo, adding a config.json and preprocessor_config.json file to the repo, then changing the wavlm_hub from "microsoft/wavlm-large" to me "myaccount_pretrained_wavlm".

Unfortunately, I have had no success in doing this, I keep getting the " the transferred parameters did not have parameters for the key: xxx" error for all parameters of the wavlm model. 

This also happens when I try to load one of SpeechBrain's repos, like "speechbrain/asr-wav2vec2-librispeech" for example. 

Any help here would be much appreciated. Thanks!

### Expected behaviour

I would expect the pretrained models from speechbrain or that I've trained using speechbrain and hosted on my own huggingface repo to download and function just like the pretrained microsoft/wavlm-large model does when I point the "wavlm_hub" to that repo.

### To Reproduce

#wavlm_hub: "microsoft/wavlm-large" # this works fine
wavlm_hub: "mtauro/wavlm_verus_k1_87"
wavlm_folder: !ref <save_folder>/wavlm_checkpoint
language: english

wavlm: !new:speechbrain.lobes.models.huggingface_wav2vec.HuggingFaceWav2Vec2
    source: !ref <wavlm_hub>
    save_path: !ref <save_folder>/wavlm_checkpoint

modules:
    wavlm: !ref <wavlm>

checkpointer: !new:speechbrain.utils.checkpoints.Checkpointer
    checkpoints_dir: !ref <save_folder>
    recoverables:
        wavlm: !ref <wavlm>

**Comment by Adel-Moumen** [comment]
Hello @miketauro,

Thanks for raising this issue. 

Could you please have a look at the Pretrainer Class section in the following tutorial https://colab.research.google.com/drive/1LN7R3U3xneDgDRK2gC5MzGkLysCWxuC3?usp=sharing#scrollTo=sNO_28zyT5OS try to reproduce the example to see if you are able to load your model. (e.g. create the model with the  "microsoft/wavlm-large" ckpt and then load the new weights). 

Best.

**Comment by miketauro** [comment]
Hi @Adel-Moumen, thanks for your response. So you're suggesting the loadable model is the microsoft/wavlm-large model, and then the pretrained path is to my own huggingface repo? I.e. 

model = HuggingFaceWav2Vec2("microsoft/wavlm-large", save_path='/content/pretrained/')
pretrain = Pretrainer(loadables={'model': model}, paths={'model': "mtauro/wavlm_verus_k1_87/wavlm.ckpt''}) 

If so, I have tried this and received the same transferred parameters error. Am I still doing something wrong?

**Comment by Adel-Moumen** [comment]
Yes so it means there's an issue with your ckpt. Hmm could you please show me your YaML file? And what is the key error that you have ?

**Comment by miketauro** [comment]
Hi Abel, I actually just ran """
from speechbrain.lobes.models.huggingface_wav2vec import HuggingFaceWav2Vec2

# HuggingFace model hub
model_hub_w2v2 = "microsoft/wavlm-large"


model = HuggingFaceWav2Vec2(model_hub_w2v2, save_path='/content/pretrained/')

from speechbrain.utils.parameter_transfer import Pretrainer

# Initialization of the pre-trainer 
pretrain = Pretrainer(loadables={'model': model}, paths={'model': 'mtauro/wavlm_verus_k1_87/wavlm.ckpt'})

# We download the pretrained model from HuggingFace in this case
pretrain.collect_files()
pretrain.load_collected(device='cpu')

"""
on colab and did not get any errors. However, when I ran my own experiment, I did not use the pretrainer, instead I simply substituted the wavlm_hub to point to my repository with my wavlm.ckpt and I got the errors that say "loading from results/wavlm_vrs_ck_iva/k1/save/wavlm_checkpoint/wavlm.ckpt, the transferred parameters did not have parameters for the key: feature_extractor.conv_layers.0.conv.weight". It appears to say this for every parameter.

I've read here (https://github.com/speechbrain/speechbrain/tree/develop/recipes/CommonVoice/self-supervised-learning/wav2vec2) that uploading a model.ckpt to a huggingface repo should work in the way I tried above but have not yet had success. I'm thinking maybe the preprocessor and config files must be modified?

**Comment by Adel-Moumen** [comment]
could you please show me the ckpt folder ?

**Comment by miketauro** [comment]
I haven't uploaded the whole checkpoint folder, just the wavlm.ckpt and .json files here (https://huggingface.co/mtauro/wavlm_verus_k1_87/tree/main). Was I supposed to upload the entire folder?
